src/sb3-serialize.ts

```diff
@@ -223,15 +225,42 @@ function deserializeBlockObject(serialized: SB3SerializedBlockObject, id: string
     return [block, ...newBlocks];
 }
 
-export function deserializeVariables(serialized: SB3Variables): Iterator<VariableProperties> {
+export function deserializeVariables(serialized: SB3ScalarVariables): Iterator<VariableProperties> {
     return map(
         Object.entries(serialized),
         ([id, [name, value, isCloud]]): VariableProperties => ({
             id,
             name,
             value,
             isCloud: Boolean(isCloud), //when  isCloud is undefined
-            type: '', // this means scalar type in https://github.com/LLK/scratch-vm/blob/33ef283787d4ea9a90c3d0d069a6b97dee24f51b/src/engine/variable.js#L49
+            type: ScratchVariableTypes.SCALAR_TYPE, // this means scalar type in https://github.com/LLK/scratch-vm/blob/33ef283787d4ea9a90c3d0d069a6b97dee24f51b/src/engine/variable.js#L49
+        }),
+    );
+}
+
+export function deserializeBroadcastVariables(
+    serialized: SB3BroadcastVariables,
+): Iterator<VariableProperties> {
+    return map(
+        Object.entries(serialized),
+        // name and value are the same for broadcasts. documented here: https://github.com/LLK/scratch-vm/blob/5adf5ba8fcbd9099a7258c810fd9099ceb39aadf/src/serialization/sb3.js#L398
+        ([id, value]): VariableProperties => ({
+            id,
+            name: value,
+            value,
+            type: ScratchVariableTypes.BROADCAST_MESSAGE_TYPE,
+        }),
```

Comment:
is
```
            name: value,
            value,
```
intentional because it looks like you have 2 of the params with the same value?

Reply:
yeah - for "broadcast message" variables - `name === value` in the VM land